sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py

```diff
@@ -427,35 +439,81 @@ def __init__(self, host, port=WEBSOCKET_PORT, connect_timeout=None, ssl=None, **
         self._custom_endpoint = kwargs.get("custom_endpoint")
         self.host = host
         self.ws = None
-        self._http_proxy = kwargs.get('http_proxy', None)
+        self.session = None
+        self._http_proxy = kwargs.get("http_proxy", None)
 
     async def connect(self):
         http_proxy_host, http_proxy_port, http_proxy_auth = None, None, None
+        ssl_opts: Optional[SSLContext] = self._build_ssl_opts(self.sslopts) if len(self.ssl_opts) > 0 else None
+
         if self._http_proxy:
-            http_proxy_host = self._http_proxy['proxy_hostname']
-            http_proxy_port = self._http_proxy['proxy_port']
-            username = self._http_proxy.get('username', None)
-            password = self._http_proxy.get('password', None)
+            http_proxy_host = self._http_proxy["proxy_hostname"]
+            http_proxy_port = self._http_proxy["proxy_port"]
+            username = self._http_proxy.get("username", None)
+            password = self._http_proxy.get("password", None)
             if username or password:
                 http_proxy_auth = (username, password)
```

Comment:
I think line 461-465 should go here somehow (another dependency try/except?) or username/password should get pulled out. username/password are declared inside if self._http_proxy statement (line 461)